From 831724db61d18f5544a40a2a80f708388f5f023b Mon Sep 17 00:00:00 2001
From: phanirithvij <phanirithvij2000@gmail.com>
Date: Thu, 20 Aug 2026 16:15:11 +0530
Subject: [PATCH] Retain query params

Signed-off-by: phanirithvij <phanirithvij2000@gmail.com>
---
 src/pages.go | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/src/pages.go b/src/pages.go
index 6e4d3df..7488b08 100644
--- a/src/pages.go
+++ b/src/pages.go
@@ -164,7 +164,9 @@ func getPage(w http.ResponseWriter, r *http.Request) error {
 			)
 			if err == nil {
 				if !hasProjectSlash {
-					writeRedirect(w, http.StatusFound, r.URL.Path+"/")
+					newURL := *r.URL
+					newURL.Path += "/"
+					writeRedirect(w, http.StatusFound, newURL.String())
 					return nil
 				}
 				sitePath, manifest, metadata = projectPath, projectManifest, projectMetadata
@@ -364,8 +366,9 @@ func getPage(w http.ResponseWriter, r *http.Request) error {
 			} else {
 				// redirect from `dir` to `dir/`, otherwise when `dir/index.html` is served,
 				// links in it will have the wrong base URL
-				newPath := r.URL.Path + "/"
-				writeRedirect(w, http.StatusFound, newPath)
+				newURL := *r.URL
+				newURL.Path += "/"
+				writeRedirect(w, http.StatusFound, newURL.String())
 				return nil
 			}
 		} else if entry.GetType() == Type_Symlink {
-- 
2.55.0

